### ISSUE TITLE
Skin Shortcuts - Unable to save menu 

### ISSUE THREAD

**Issue by Travion1976** [drinfernoo/plugin.program.autowidget#32]
Tested on skin Arctic Zephyr 2 and also Aura Mod
Device: Tested on Mac Kodi 18.7 and also VMWareFusion Libreelec Kodi 18.7

When editing shortcuts -> configuring the widgets -> Choose widget -> AutoWidgets -> select and -> setting a label all these steps work fine.
When trying to change the Features under Widgets such as Widget Style, Content Type, Sort By, sort Direction 
I run into the error "Skin Shortcuts - Unable to save menu"
I have done some research and found out that is has something to do with the settings saved in script.skinshortcuts but I am not able to fix it. 

My Kodi Log Viewer does not show any errors concerning this.
Would you have any advice, or point me in a direction I could look at?

Thanks

PS I love your AutoWidget add on ! Thank you so much for this amazing tool ! 

![Screenshot 2020-06-23 at 09 47 53](https://user-images.githubusercontent.com/50074166/85375541-bfdcc700-b536-11ea-8795-672aa866050e.png)
![Screenshot 2020-06-23 at 09 47 57](https://user-images.githubusercontent.com/50074166/85375542-c0755d80-b536-11ea-8da0-94c84a890e5d.png)

**Comment by drinfernoo** [comment]
This happens because of a parsing error while processing the `skin.name.here.properties` file, causing the file to get written back incorrectly... and I had previously had quite a bit of trouble working around it.

Fortunately, I'm well on the way to a 2.0 release, which will fix this as well 👍

**Comment by Travion1976** [comment]
Thank you so much for you reply ! That sounds awesome that you have a fix on the way.
Is there anything I can do in the meantime or is the trouble not worth it ?
Thanks again for your hard work.
Best regards

**Comment by drinfernoo** [comment]
**If** you feel like trying to fix the error manually inside the file, then you could, but it's a bit tricky, and will be different for every user. 

EDIT: For what it's worth, when using a debug log, I believe Skin Shortcuts does give more information regarding this, but it's still not exactly straightforward to fix by hand.

**Comment by Travion1976** [comment]
Ok great , I will take a look at this file and see if I can figure something out. If this gets too complicated I will look forward to see your 2.0 release.

**Comment by drinfernoo** [comment]
I decided to go ahead and recreate it so that I can give a bit of guidance for those that are running into this. For what it's worth, I believe this should only happen when using these in "non-standard" places, likes hubs or possibly in some skins' "spotlight" area.

So, I assigned an AutoWidget to a submenu hub widget in Arctic Zephyr 2, initialized the widget (also going to be a thing of the past in 2.0 😁), and then returned to the Skin Shortcuts dialog to receive the same error as reported above.

The contents of the file at `special://userdata/addon_data/script.skinshortcuts/skin.arctic.zephyr.2.properties` follow:
```python
[['2.1', 'popular', 'widgetTarget', u'videos'],
 ['2.1', 'popular', 'icon', 'DefaultShortcut.png'],
 ['2.1', 'popular', 'thumb', 'image://C%3a%5cUsers%5cdrinfernoo%5cDesktop%5ckodi%5cLeia%2018.6%20Testing%20Again%5cportable_data%5caddons%5cplugin.video.themoviedb.helper%2fresources%2ficons%2ftmdb%2fpopular.png/'],
 ['2.1', 'randompathfrommovies-1952728a-2db1-45ef-a6f2-8f1dcb70bae5', 'translatedPath', 'ActivateWindow($INFO[Skin.String(autowidget-1952728a-2db1-45ef-a6f2-8f1dcb70bae5-target)],
$INFO[Skin.String(autowidget-1952728a-2db1-45ef-a6f2-8f1dcb70bae5-action)],
return)'],
 ['2.1', 'randompathfrommovies-1952728a-2db1-45ef-a6f2-8f1dcb70bae5', 'widgetTarget', u'videos'],
 ['2.1', 'randompathfrommovies-1952728a-2db1-45ef-a6f2-8f1dcb70bae5', 'icon', 'DefaultShortcut.png'],
 ['2.1', 'randompathfrommovies-1952728a-2db1-45ef-a6f2-8f1dcb70bae5', 'thumb', 'image://C%3a%5cUsers%5cdrinfernoo%5cDesktop%5ckodi%5cLeia%2018.6%20Testing%20Again%5cportable_data%5caddons%5cplugin.program.autowidget%5cresources%5cmedia%5cicon%5csync.png/'],
 ['20342.1', 'season3', 'widgetTarget', u'videos'],
 ['20342.1', 'season3', 'widgetAspect', u'Poster'],
 ['20342.1', 'season3', 'icon', 'DefaultShortcut.png'],
 ['20342.1', 'season3', 'thumb', 'image://https%3a%2f%2fimage.tmdb.org%2ft%2fp%2fw500%2fjjHt0ebmQm4wySfXYuQnVAYLMO7.jpg/']]
```

As I mentioned previously, this file will look different depending on the users' defined widgets, but the bit you're looking for (in the snippet above, it's lines 4-6) is wherever there is that "rogue" bit of data. each line in this file should consist of a list of four string elements: the "menu" ID from Skin Shortcuts, the "group" ID, an attribute ID, and the value of that attribute.

```python
['2.1', 'randompathfrommovies-1952728a-2db1-45ef-a6f2-8f1dcb70bae5', 'translatedPath', 'ActivateWindow($INFO[Skin.String(autowidget-1952728a-2db1-45ef-a6f2-8f1dcb70bae5-target)],
$INFO[Skin.String(autowidget-1952728a-2db1-45ef-a6f2-8f1dcb70bae5-action)],
return)'],
```

And what's happening is that this is getting "split" when written to the file, and "lost in translation". For these, you basically just want to make it be all on one line again, by removing the line breaks:

```python
['2.1', 'randompathfrommovies-1952728a-2db1-45ef-a6f2-8f1dcb70bae5', 'translatedPath', 'ActivateWindow($INFO[Skin.String(autowidget-1952728a-2db1-45ef-a6f2-8f1dcb70bae5-target)],$INFO[Skin.String(autowidget-1952728a-2db1-45ef-a6f2-8f1dcb70bae5-action)],return)'],
```

That should allow Skin Shortcuts to rebuild the menu correctly 😁 

As I mentioned previously, I haven't figured out an exact fix for this issue, but in future versions, it will be a moot point 👌

**Comment by Travion1976** [comment]
Oh man, thank you so much for this help. I will take a look at it later on today and give feedback if this worked.

**Comment by Travion1976** [comment]
Correcting the line break definitely works. It only works once, so if you need to correct multiple widgets and menus, you need to go back again and correct that same line break after each change. 
This is definitely a good work around. Thanks again !

**Comment by drinfernoo** [comment]
I was just reminded that I never closed this after 2.0 came out!
